x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -103,18 +112,32 @@ synchronized void updatePartialResponse(
         this.totalHits = totalHits;
         this.reducedAggsSource = reducedAggs;
         this.reducePhase = reducePhase;
+        if (isFinalLocalReduce && clusters.getTotal() > (clusters.getSuccessful() + clusters.getSkipped())) {
+            // currently only the ccsMinimizeRoundTrip=true creates Clusters in their initial state (where successful=0)
```

Comment:
would it be more explicit to rely on the clusters.isMinimizeRoundtrips here instead of on the count of the clusters to guess whether the object needs updating or not?

Reply:
For now that would probably work (would have to test to be sure of edge cases), but it would then lock in the functionality to only isMinimizeRoundtrips, but maybe that's better for now. I'll test it.

Reply:
Looks good. Fixed.